pkgs/development/tools/database/dynein/default.nix

```diff
@@ -0,0 +1,47 @@
+{ fetchFromGitHub
+, lib
+, Security
+, openssl
+, pkg-config
+, rustPlatform
+, stdenv
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "dynein";
+  version = "0.2.1";
+
+  src = fetchFromGitHub {
+    owner = "awslabs";
+    repo = "dynein";
+    rev = "v${version}";
+    sha256 = "sha256-QhasTFGOFOjzNKdQtA+eBhKy51O4dFt6vpeIAIOM2rQ=";
+  };
+
+  # Use system openssl.
+  OPENSSL_NO_VENDOR = 1;
+
+  cargoHash = "sha256-QyhoYgqBfK6LCdtLuo0feVCgIMPueYeA8MMGspGLbGQ=";
+
+  nativeBuildInputs = [ pkg-config ];
+
+  buildInputs = [
+    openssl
+  ] ++ lib.optionals stdenv.isDarwin [ Security ];
+
+  preBuild = ''
+    export OPENSSL_DIR=${lib.getDev openssl}
+    export OPENSSL_LIB_DIR=${lib.getLib openssl}/lib
+  '';
+
+  # Tests are long to compile
+  doCheck = false;
```

Comment:
To be honest I don't think that's a good reason for disabling tests.

Reply:
The test even fail when just enabling them :(
![image](https://github.com/NixOS/nixpkgs/assets/80165193/2e00ed35-9c7b-412e-ad58-7896d9776b71)

Reply:
Do we need an instance of dynamodb for these tests?

Reply:
So I set the tests to run, but only the `cli_tests`. The integration suite starts downloading DynamoDB with Docker, which will naturally lead to chaos and sadness from the nix build context.